### PR TITLE
chore(docs): generate CRDs API reference in markdown

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

This adds a Makefile target that generates a CRDs reference in Markdown using [github.com/elastic/crd-ref-docs](github.com/elastic/crd-ref-docs) tool. That can be used as a replacement for https://docs.konghq.com/kubernetes-ingress-controller/latest/references/custom-resources/. 

It also tries to make godoc strings unified across the API packages (beginning with a capital, ending with a period, etc.).

For now, I just copy-pasted it into the docs repository ([docs preview](https://deploy-preview-4813--kongdocs.netlify.app/kubernetes-ingress-controller/latest/references/custom-resources/)). I'm waiting for a response to my question about how the docs team would prefer this to be included in the docs build.

**Which issue this PR fixes**:

Part of #3019. 

**Special notes for your reviewer**:

It still needs verification whether we're not losing some valuable content in comparison to the existing CRDs reference in the docs.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

~- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR~

### CODE REVIEW

**Review by randmonkey — COMMENTED** [review]
Compared to current[ CRD page](https://docs.konghq.com/kubernetes-ingress-controller/latest/references/custom-resources/) in docs , this generated markdown is only telling the fields of CRDs. I think it is better to remain the original CRD page for introduction of CRDs and examples, and this generated page as a reference.

**Comment by czeslavo** [comment]
> I really like the idea of having generated docs (and the fact that we could automate it) but OTOH I agree with @randmonkey regarding the fact that at the current stage the generate page lacks a bit more context (unless we only want it to serve as a reference, contrary to what our current page for CRDs https://docs.konghq.com/kubernetes-ingress-controller/latest/references/custom-resources/ does, which is to add descriptions and examples to some entries).
> 
> Do you think we could add those types of descriptions here @czeslavo? Or perhaps we could strip what we currently have from the "reference" bits and make it a separate page from the one proposed in this PR?

Yes, that's what I meant in the PR's notes to reviewers. As there is some content that is more in a tutorial/example style in the former reference, I would prefer to extract these examples and make them a separate article, e.g. `Using Custom Resource Definitions`.

**Comment by pmalek** [comment]
> Yes, that's what I meant in the PR's notes to reviewers. As there is some content that is more in a tutorial/example style in the former reference, I would prefer to extract these examples and make them a separate article, e.g. `Using Custom Resource Definitions`.

Cool. What I'd suggest is to make those 2 interlinked or collapsable into the same section (at docs.konghq.com)

File: docs/api-reference.md [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+
+| Field | Description |
+| --- | --- |
+| `apiVersion` _string_ | `configuration.konghq.com/v1alpha1`
+| `kind` _string_ | `IngressClassParameters`
+| `metadata` _[ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#objectmeta-v1-meta)_ | Refer to Kubernetes API documentation for fields of `metadata`. |
+| `spec` _[IngressClassParametersSpec](#ingressclassparametersspec)_ |  |
+
+
+### IngressClassParametersList
```
[the remark this answers is not in the document]
Comment (czeslavo) [reply docs/api-reference.md]: Agree: [35f082b](https://github.com/Kong/kubernetes-ingress-controller/pull/3185/commits/35f082b17405db5f2d5f67c92c2023864b24dede)

File: docs/api-reference.md [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+- [KongConsumer](#kongconsumer)
+- [KongConsumerList](#kongconsumerlist)
+- [KongIngress](#kongingress)
+- [KongIngressList](#kongingresslist)
+- [KongPlugin](#kongplugin)
+- [KongPluginList](#kongpluginlist)
+
+
+
+### ConfigSource
```
[the remark this answers is not in the document]
Comment (czeslavo) [reply docs/api-reference.md:18]: Yup, it is: [c60b9ad](https://github.com/Kong/kubernetes-ingress-controller/pull/3185/commits/c60b9ade2fadf01b3d4586850fb7ba87847c72a4) :)

**Review by randmonkey — APPROVED** [review]

**Pull request merged**